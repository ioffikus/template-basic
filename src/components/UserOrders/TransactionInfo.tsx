import React from 'react';

interface IProps {
  ui: Alicanto.Models.UI;
  details: string;
  status: string;
}

interface IState {}

class TransactionInfo extends React.Component<IProps, IState> {
  render() {
    const transactionUI = this.props.ui.routes.user.orders.transaction;
    const { details, status } = this.props;
    return (
      <div className="root">
        <style jsx>{`
          .root {
            font-size: 10px;
            font-weight: 500;
          }
          .name {
            margin-right: 10px;
            color: #b2bdcb;
            font-size: 9px;
          }
        `}</style>
        {status && (
          <div>
            <span className="name">{transactionUI.status.i18n.defaultText} : </span>
            <span>{status}</span>
          </div>
        )}
        {details && (
          <div>
            <span className="name">{transactionUI.details.i18n.defaultText} : </span>
            <span>{details}</span>
          </div>
        )}
      </div>
    );
  }
}

export default TransactionInfo;
