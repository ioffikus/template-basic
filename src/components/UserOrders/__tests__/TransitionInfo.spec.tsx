import React from 'react';
import { shallow } from 'enzyme';
import uiStateMock from 'src/store/__mocks__/uiStore';
import TransactionInfo from '../TransactionInfo';

describe(null, () => {
  it('render() correct with status and details', () => {
    const status = 'dymmy status';
    const details = 'dymmy details';
    const wrapper = shallow(<TransactionInfo status={status} details={details} ui={uiStateMock} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render() correct with status and empty details', () => {
    const status = 'dymmy status';
    const details = '';
    const wrapper = shallow(<TransactionInfo status={status} details={details} ui={uiStateMock} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render() correct with details and empty status', () => {
    const status = '';
    const details = 'dymmy details';
    const wrapper = shallow(<TransactionInfo status={status} details={details} ui={uiStateMock} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render() correct with empty status and details', () => {
    const status = '';
    const details = '';
    const wrapper = shallow(<TransactionInfo status={status} details={details} ui={uiStateMock} />);
    expect(wrapper).toMatchSnapshot();
  });
});
