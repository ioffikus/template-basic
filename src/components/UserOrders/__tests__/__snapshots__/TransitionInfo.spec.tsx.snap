// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() correct with details and empty status 1`] = `
<div
  className="jsx-1430030103 root"
>
  <JSXStyle
    css=".root.jsx-1430030103{font-size:10px;font-weight:500;}.name.jsx-1430030103{margin-right:10px;color:#b2bdcb;font-size:9px;}"
    styleId="1430030103"
  />
  <div
    className="jsx-1430030103"
  >
    <span
      className="jsx-1430030103 name"
    >
      Transaction Details
       : 
    </span>
    <span
      className="jsx-1430030103"
    >
      dymmy details
    </span>
  </div>
</div>
`;

exports[`render() correct with empty status and details 1`] = `
<div
  className="jsx-1430030103 root"
>
  <JSXStyle
    css=".root.jsx-1430030103{font-size:10px;font-weight:500;}.name.jsx-1430030103{margin-right:10px;color:#b2bdcb;font-size:9px;}"
    styleId="1430030103"
  />
</div>
`;

exports[`render() correct with status and details 1`] = `
<div
  className="jsx-1430030103 root"
>
  <JSXStyle
    css=".root.jsx-1430030103{font-size:10px;font-weight:500;}.name.jsx-1430030103{margin-right:10px;color:#b2bdcb;font-size:9px;}"
    styleId="1430030103"
  />
  <div
    className="jsx-1430030103"
  >
    <span
      className="jsx-1430030103 name"
    >
      Transaction Status
       : 
    </span>
    <span
      className="jsx-1430030103"
    >
      dymmy status
    </span>
  </div>
  <div
    className="jsx-1430030103"
  >
    <span
      className="jsx-1430030103 name"
    >
      Transaction Details
       : 
    </span>
    <span
      className="jsx-1430030103"
    >
      dymmy details
    </span>
  </div>
</div>
`;

exports[`render() correct with status and empty details 1`] = `
<div
  className="jsx-1430030103 root"
>
  <JSXStyle
    css=".root.jsx-1430030103{font-size:10px;font-weight:500;}.name.jsx-1430030103{margin-right:10px;color:#b2bdcb;font-size:9px;}"
    styleId="1430030103"
  />
  <div
    className="jsx-1430030103"
  >
    <span
      className="jsx-1430030103 name"
    >
      Transaction Status
       : 
    </span>
    <span
      className="jsx-1430030103"
    >
      dymmy status
    </span>
  </div>
</div>
`;
